File: algorithms/go/1736-latest-time-by-replacing-hidden-digits/latest-time-by-replacing-hidden-digits.go
package problem1736

import "fmt"

// maximumTime 贪心
func maximumTime(time string) string {
	t := []byte(time)
	if t[0] == '?' {
		if '4' <= time[1] && time[1] <= '9' {
			t[0] = '1'
		} else {
			t[0] = '2'
		}
	}
	if t[1] == '?' {
		if t[0] == '2' {
			t[1] = '3'
		} else {
			t[1] = '9'
		}
	}
	if t[3] == '?' {
		t[3] = '5'
	}
	if t[4] == '?' {
		t[4] = '9'
	}
	return string(t)
}

func main() {
	time := "23:23"
	fmt.Println([]byte(time))
}

File: algorithms/python/1736_latest_time_by_replacing_hidden_digits/latest_time_by_replacing_hidden_digits.py
"""
1736. latest time by replacing hidden digits

https://leetcode.com/problems/latest-time-by-replacing-hidden-digits/
https://leetcode-cn.com/problems/latest-time-by-replacing-hidden-digits/

替换隐藏数字得到的最晚时间
"""


def maximum_time(time: str) -> str:
    """
    贪心
    """
    time = list(time)
    if time[0] == '?':
        time[0] = '1' if '4' <= time[1] <= '9' else '2'
    if time[1] == '?':
        time[1] = '3' if time[0] == '2' else '9'
    if time[3] == '?':
        time[3] = '5'
    if time[4] == '?':
        time[4] = '9'
    return ''.join(time)
